### DISCUSSION TITLE
pivot/object origin snapping

### DISCUSSION THREAD

**Discussion by alFrame** [JulienHeijmans/quicksnap#10] [Ideas]
Coming from Maya, this looks like a time saver. I bought it over on Blender Market.

Do you think it would be tough to make it so we could snap the pivot of an object to verts, edges, faces etc. and also axis constrained?

Sometime, well, may times, I need to adjust the pivot and then snap an object to something.

Edit: I should have tried it before I wrote. It's awesome. My muscle memory is fighting me like nobodies business ever since I started using Blender. Especially in the viewport. And tbh. that's an area where I still pursue to replicate Maya in Blender. Besides, I am using Blender setup and not industry standard. For one, it's tough to find tutorials when using something else than standard mode and, I think it's also about honoring the thinking behind a piece of software. Always!

Long story short, I mapped "Affect Only" to "Origin", using the "insert" key on the keyboard. Then I use eye balling, or the regular Blender snapping to snap the pivot.

So, technically all is good. But still I think using one metaphor to move "stuff" would be very cool!

**Comment by JulienHeijmans** [comment]
Hey!
Sooo, at the end of the day, you managed to do what you wanted? :)

**Reply by alFrame** [reply]
Yeah, sort of. Still would be great if your tool could also move/snap the pivot.

Then we could use the same shortcuts and gestures to move/snap the pivot and don't have to use the Blender default snapping metaphor.

**Reply by JulienHeijmans** [reply]
Aah alright, so you would like to be able to move the pivot/origin of an object while preserving the vertices in place.

How would you imagine it work? A different hotkey to enable the tool in that 'Pivot move' mode, a modifier key while you are moving the Pivot, or something like that?

**Reply by alFrame** [reply]
Exactly!

I imagine invoking the quick snap with ctrl-shift-alt-V for instance and then exactly the same as with objects.

**Reply by JulienHeijmans** [reply]
Alright, I will add it to the list of future features.

**Reply by alFrame** [reply]
That's awesome! Thanks for considering it!
